connectors/es/index.py

```diff
@@ -49,17 +57,25 @@ async def fetch_by_id(self, doc_id):
             logger.critical(f"The server returned {e.status_code}")
             logger.critical(e.body, exc_info=True)
             if e.status_code == 404:
-                return None
+                raise DocumentNotFoundError(
+                    f"Couldn't find document in {self.index_name} by id {doc_id}"
+                )
             raise
 
-        return self._create_object(resp.body)
+        return resp.body
 
     async def index(self, doc):
         resp = await self.client.index(index=self.index_name, document=doc)
         return resp["_id"]
 
-    async def update(self, doc_id, doc):
-        await self.client.update(index=self.index_name, id=doc_id, doc=doc)
+    async def update(self, doc_id, doc, if_seq_no=None, if_primary_term=None):
```

Comment:
So for now `if_seq_no` is not used, should it be added in a different PR, or should logic that uses it be added here?

Reply:
I'm trying to break the changes as small as possible. The purpose of this PR is, as the title says, to `Add concurrency control functionality`, i.e. make it capable of using optimistic concurrency control, but not necessarily use it.